android/src/main/java/com/reactnative/googlecast/MediaInfoBuilder.java

```diff
@@ -95,14 +99,38 @@
 
         Boolean isLive = ReadableMapUtils.getBoolean(parameters, "isLive");
         int streamType =
-                isLive != null && isLive
-                        ? MediaInfo.STREAM_TYPE_LIVE
-                        : MediaInfo.STREAM_TYPE_BUFFERED;
+            isLive != null && isLive
+                ? MediaInfo.STREAM_TYPE_LIVE
+                : MediaInfo.STREAM_TYPE_BUFFERED;
+
+        List mediaTracks = new ArrayList();
+        if (parameters.hasKey("tracks") && parameters.getArray("tracks") != null) {
+            ReadableArray tracks = parameters.getArray("tracks");
+
+            for (int i = 0; i < tracks.size(); i++) {
+                ReadableMap trackData = tracks.getMap(i);
+
+                MediaTrack track = new MediaTrack.Builder(i, MediaTrack.TYPE_TEXT)
+                    .setName(trackData.getString("title"))
+                    .setSubtype(MediaTrack.SUBTYPE_SUBTITLES)
+                    .setContentId(trackData.getString("id"))
```

Comment:
According with the documentation `setContentId` should be the url for the `vtt`, and then you could use `uri` instead `id` this can be confuse for the developers:

```suggestion
                    .setContentId(trackData.getString("uri"))
```

https://developers.google.com/cast/docs/android_sender/media_tracks#configure_a_track

Reply:
Actually where i implemented my fork the id contains the uri ;) Maybe if there is any change this will merge one day i can update this PR to align with yours.